### ISSUE TITLE
Impossible to check null session without getSession warning

### ISSUE THREAD

**Issue by jdgamble555** [supabase/supabase-js#1707]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

This is an `auth.js` issue related to supabase/auth-js#873, so I'm not going to link the other repos.

It is IMPOSSIBLE to check for a session without contacting the Supabase server or displaying the annoying warning.

To be more specific, if my session is NULL, there is NO reason for me to need to contact the Supabase server. This means, for my pages that don't require a login, it will ALWAYS contact the Supabase server needlessly.

The problem lies in the [suppressGetSessionWarning](https://github.com/supabase/auth-js/blob/master/src/GoTrueClient.ts#L1116) can only be called when the [_saveSession](https://github.com/supabase/auth-js/blob/master/src/GoTrueClient.ts#L2033) gets called. What if there is no session? Then we have to use `getUser()` for no reason, or deal with a needless warning OVER and OVER.

## To Reproduce

Check a user's session with `getSession()` while not logged in, you get the warning.

## Expected behavior

It is expected that we can use `getSession()` without a warning. The warning only shows up when there is a session and we can use `getSession()` without calling `getUser()`.

## Other Problems

This has been mentioned in all three github issues, but still not solved.

Please do not close this issue until both the NextJS and SvelteKit (preferably all docs) have been updated to NOT only use `getUser()` on the server for NULL.

Adding an extra needless fetch is a big deal.

- https://supabase.com/docs/guides/getting-started/tutorials/with-nextjs
- https://supabase.com/docs/guides/getting-started/tutorials/with-sveltekit

J

**Comment by mandarini** [comment]
@jdgamble555 is this issue still valid?

**Comment by jdgamble555** [comment]
I think this can be closed... we just need to the docs updated (particularly for NextJS and SvelteKit).

Thanks,

J

**Comment by mandarini** [comment]
Yep, the docs work is under way! It's being tracked in the other issue with @ChrisChinchilla !

**Comment by mandarini** [comment]
Thank you @jdgamble555 for the valuable contributions!
